### PR TITLE
Add reproduction for "Closing Drop-down in Join column selector causes non-drop-down columns to disappear"

### LINKED ISSUE #19894: Closing Drop-down in Join column selector causes non-drop-down columns to disapper
To repro:

**To Reproduce**
1. Simple question > Sample Dataset > Products - summarize Count group by Category - save as "Q1"
2. Simple question > Sample Dataset > Products - summarize Sum of Price group by Category - save as "Q2"
3. Simple question > Sample Dataset > Products - summarize Average of Rating group by Category - save as "Q3"
4a. Custom question > Saved Questions > "Q1" - join "Q2" on Category
4b. Add another join against "Q3" on Category. When selecting the columns expand the `Question ___` dropdown. Other columns are no longer visible. Close the dropdown. No columns are visible at all.

bye

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/19894

### Description

The the issue is fixed in v50, this PR adds a reproduction for it.

[Stress test](https://github.com/metabase/metabase/actions/runs/9681681744)

### CODE REVIEW

File: e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js [hunk 1]
```diff
@@ -994,3 +994,76 @@ describe("issue 40399", () => {
...
+    createQuestion({
+      name: "Q3",
+      query: {
+        "source-table": PRODUCTS_ID,
+        aggregation: [["avg", ["field", PRODUCTS.RATING, null]]],
+        breakout: [["field", PRODUCTS.CATEGORY, null]],
+      },
+    });
+
+    cy.visit(`/question/notebook`);
```
Comment (ranquild) [inline e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js:1037]: `startNewQuestion` or something similar

**Review by ranquild — APPROVED** [review]
One comment

File: e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js [hunk 2]
```diff
@@ -994,3 +994,76 @@ describe("issue 40399", () => {
...
+    popover().findByText("Category").should("be.visible");
+    popover().findByText("Count").should("be.visible");
+
+    popover().findByText("Q1").click();
+    popover().findByText("Q2").click();
+
+    popover().findByText("Category").should("be.visible");
+    popover().findByText("Sum of Price").should("be.visible");
+
+    popover().findByText("Q1").click();
```
Comment (uladzimirdev) [inline e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js:1064]: why not `popover().within` everywhere?

Comment (romeovs) [reply e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js:1064]: These are not the same popover as a new one opens for each column.

**Pull request merged**